src/app/zap-templates/zcl/data-model/chip/switch-cluster.xml

```diff
@@ -71,63 +74,63 @@ Interactions with the switch device are exposed as attributes (for the latching
-      <description>InitialPress</description>
-      <field id="0" name="NewPosition" type="int8u" />
+
+    <event side="server" code="0x01" priority="info" name="InitialPress">
+      <description>This event SHALL be generated, when the momentary switch starts to be pressed (after debouncing).</description>
+      <field id="0" name="NewPosition" type="int8u" max="254"/>
       <mandatoryConform>
         <feature name="MS"/>
       </mandatoryConform>
     </event>
-    <event side="server" code="0x02" priority="info" name="LongPress" optional="true">
-      <description>LongPress</description>
-      <field id="0" name="NewPosition" type="int8u" />
+
+    <event side="server" code="0x02" priority="info" name="LongPress">
+      <description>This event SHALL be generated when the momentary switch has been pressed for a &quot;long&quot; time.</description>
+      <field id="0" name="NewPosition" type="int8u" max="254"/>
       <mandatoryConform>
         <feature name="MSL"/>
       </mandatoryConform>
     </event>
-    <event side="server" code="0x03" priority="info" name="ShortRelease" optional="true">
-      <description>ShortRelease</description>
-      <field id="0" name="PreviousPosition" type="int8u" />
+
+    <event side="server" code="0x03" priority="info" name="ShortRelease">
+      <description>If the server has the Action Switch (AS) feature flag set, this event SHALL NOT be generated at all, since setting the Action Switch feature flag forbids the Momentary Switch ShortRelease (MSR) feature flag from being set.</description>
+      <field id="0" name="PreviousPosition" type="int8u" max="254"/>
       <mandatoryConform>
         <feature name="MSR"/>
       </mandatoryConform>
     </event>
-    <event side="server" code="0x04" priority="info" name="LongRelease" optional="true">
-      <description>LongRelease</description>
-      <field id="0" name="PreviousPosition" type="int8u" />
+
+    <event side="server" code="0x04" priority="info" name="LongRelease">
+      <description>This event SHALL be generated, when the momentary switch has been released (after debouncing) and after having been pressed for a long time, i.e. this event SHALL be generated when the switch is released if a LongPress event has been generated since the previous InitialPress event.</description>
+      <field id="0" name="PreviousPosition" type="int8u" max="254"/>
       <mandatoryConform>
         <feature name="MSL"/>
       </mandatoryConform>
     </event>
-    <event side="server" code="0x05" priority="info" name="MultiPressOngoing" optional="true">
-      <description>MultiPressOngoing</description>
-      <field id="0" name="NewPosition" type="int8u" />
-      <field id="1" name="CurrentNumberOfPressesCounted" type="int8u" />
+
+    <event side="server" code="0x05" priority="info" name="MultiPressOngoing">
+      <description>If the server has the Action Switch (AS) feature flag set, this event SHALL NOT be generated at all.</description>
```

Comment:
Same here, this is not the description of the event, it's the description of a constraint on the event.

Reply:
Yep, it is again from the first paragraph of the spec description. The second one actually seems to be more useful for this but not sure if Alchemy will be able to make this kind of "judgements".